### PR TITLE
Add a dynamic timeout option

### PR DESCRIPTION
**Pull request by matthew-m-hall**
Signed-off-by: Matthew Hall <matthew.hall@hpe.com>

### CODE REVIEW

**Comment by dajose** [comment]
Please provide a description of your need and why your solution attacks that problem

**Comment by matthew-m-hall** [comment]
This is to support testing at scale and avoid many updates to many libraries. As we increase the scale of configurations in our testing, commands are taking longer to output to the console,

I filed a CR asking for the ability to use a pytest marker to request the framework to set a higher timeout when creating the pexpect spawn. Esteban instead suggested that we make the timeout dynamic, so it waits for the command to finish its output before timing out for not seeing the prompt.

**Comment by matthew-m-hall** [comment]
Updated commit from feedback, submitting change as an option. The default behavior remains unchanged

File: lib/topology/platforms/shell.py [hunk 1]
```diff
@@ -453,7 +454,8 @@ def _get_connection(self, connection=None):
     def send_command(
         self, command,
         matches=None, newline=True,
-        timeout=None, connection=None, silent=False, control=False
+        timeout=None, connection=None, silent=False, control=False,
+        continuous_timeout=False
```
Comment (dajose) [inline lib/topology/platforms/shell.py]: please document this new parameter

Comment (matthew-m-hall) [reply lib/topology/platforms/shell.py]: param doc added, thx

**Pull request merged**